.github/workflows/validate.yml

```diff
@@ -11,58 +11,54 @@ jobs:
     strategy:
       matrix:
         BUILD_TYPE: [Release, Debug]
+        DISTRO: ['fedora', 'fedora_rawhide', 'fedora_mpich', 'opensuse', 'ubuntu', 'ubuntu_devel', 'ubuntu_rolling', 'ubuntu_mpich']
+        CXX: ['g++', 'clang++']
+        EXTERNAL_MPI4PY: ['On','Off']
```

Comment:
Goal should be to get rid of this switch entirely. Either we should stick to a preinstalled one if we do not have special requirements or we should always use ours, if we have special requirements. I do not see a point in having it switchable.

Reply:
It seems to currently work with all distribution packages, so we should just drop our bundled version.

Reply:
I `mpi4py` built against `mpi`? If so, can we ensure that this is always correct? Also on clusters with multiple `mpi` versions?

Reply:
Well I think that is the same with all of our other dependencies, e.g. hdf5, boost-mpi.

Reply:
On a cluster I would use spack install espressopp, but the package needs an update too.

Reply:
Hmm, for hdf5 and boost you typically can load special packages for each `mpi` version. I have not seen this for `python`.

Reply:
spack has support to load modules, but it can also build everything if e.g. the boost has not enable python.